octoprint_mrbeam/iobeam/dust_manager.py

```diff
@@ -513,13 +517,116 @@ def _validate_values(self):
                 rpm=self._rpm,
                 dust=self._dust,
                 connected=self._connected,
-                age=(time.time() - self._data_ts),
+                age=(monotonic_time() - self._data_ts),
             )
             self._pause_laser(trigger="Air filter not connected.", log_message=msg)
 
         # TODO: check for error case in connected val (currently, connected == True/False/None)
+        self._check_and_report_error()
         return result
 
+    def _check_if_one_value_is_none(self):
+        """
+        Checks if one of the values is None and sets the _fan_data_missing_ts flag.
+
+        Returns: (bool, list): True if one of the values is None, False otherwise. And List of error messages.
+        """
+        errs = []
+        result = False
+        if self._state is None or self._rpm is None or self._dust is None:
+            result = True
+            if self._fan_data_missing_ts is None:
+                self._fan_data_missing_ts = monotonic_time()
+                self._fan_data_missing_reported = False
+
+            if self._state is None:
+                errs.append("fan state:{}".format(self._state))
+
+            if self._dust is None:
+                errs.append("dust:{}".format(self._dust))
+
+            if self._rpm is None:
+                errs.append("rpm:{}".format(self._rpm))
+        else:
+            # reset counter or timer
+            self._fan_data_missing_ts = None
+        return result, errs
+
+    def _check_if_fan_is_spinning_during_a_running_job(self):
```

Comment:
This method is way too long and has possible duplications. Better move the specific error reportings into separate methods and maybe some can use the same method (not sure as there are too many parameters maybe??)

Reply:
the cognitive complexity is just 3, I would keep it as it is